recipes/bwa-mem2/simde.patch

```diff
@@ -0,0 +1,251 @@
++#define SIMDE_ENABLE_NATIVE_ALIASES
++#include <simde/x86/avx2.h>
+ #include <immintrin.h>
+ #endif
+ 
+diff --git a/src/main.cpp b/src/main.cpp
+index 3a269e1..bcc2c22 100644
+--- a/src/main.cpp
++++ b/src/main.cpp
+@@ -85,6 +85,10 @@ int main(int argc, char* argv[])
+         fprintf(stderr, "Executing in SSE4.2 mode!!\n");
+ #elif __SSE4_1__
+         fprintf(stderr, "Executing in SSE4.1 mode!!\n");        
++#elif __SSE2__
++        fprintf(stderr, "Executing in SSE2 mode!!\n");
++#else
++        fprintf(stderr, "Executing in Scalar mode!!\n");
+ #endif
+         fprintf(stderr, "-----------------------------\n");
+ 
+diff --git a/src/runsimd.cpp b/src/runsimd.cpp
+index 03c0e91..2f1190d 100644
+--- a/src/runsimd.cpp
++++ b/src/runsimd.cpp
+@@ -61,7 +61,7 @@ void __cpuidex(int cpuid[4], int func_id, int subfunc_id)
+ 	__asm__ volatile ("cpuid"
+ 			: "=a" (cpuid[0]), "=b" (cpuid[1]), "=c" (cpuid[2]), "=d" (cpuid[3])
+ 			: "0" (func_id), "2" (subfunc_id));
+-#else // on 32bit, ebx can NOT be used as PIC code
++#elif defined(__i386__)
+ 	__asm__ volatile ("xchgl %%ebx, %1; cpuid; xchgl %%ebx, %1"
+ 			: "=a" (cpuid[0]), "=r" (cpuid[1]), "=c" (cpuid[2]), "=d" (cpuid[3])
+ 			: "0" (func_id), "2" (subfunc_id));
+diff --git a/src/utils.h b/src/utils.h
+index 54a062a..144c2a0 100644
+--- a/src/utils.h
++++ b/src/utils.h
+@@ -66,6 +66,46 @@ static inline unsigned long long __rdtsc(void)
+ #endif
+ #endif
+ 
++// Apache 2.0 license
++#if defined(__aarch64__)
++  // System timer of ARMv8 runs at a different frequency than the CPU's.
++  // The frequency is fixed, typically in the range 1-50MHz.  It can be
++  // read at CNTFRQ special register.  We assume the OS has set up
++  // the virtual timer properly.
++static inline unsigned long long __rdtsc(void)
++{
++  int64_t virtual_timer_value;
++  asm volatile("mrs %0, cntvct_el0" : "=r"(virtual_timer_value));
++  return virtual_timer_value;
++}
++#elif defined(__ARM_ARCH)
++  // V6 is the earliest arch that has a standard cyclecount
++  // Native Client validator doesn't allow MRC instructions.
++#if (__ARM_ARCH >= 6)
++static inline unsigned long long __rdtsc(void)
++{
++  uint32_t pmccntr;
++  uint32_t pmuseren;
++  uint32_t pmcntenset;
++  // Read the user mode perf monitor counter access permissions.
++  asm volatile("mrc p15, 0, %0, c9, c14, 0" : "=r"(pmuseren));
++  if (pmuseren & 1) {  // Allows reading perfmon counters for user mode code.
++    asm volatile("mrc p15, 0, %0, c9, c12, 1" : "=r"(pmcntenset));
++    if (pmcntenset & 0x80000000ul) {  // Is it counting?
++      asm volatile("mrc p15, 0, %0, c9, c13, 0" : "=r"(pmccntr));
++      // The counter is set up to count every 64th cycle
++      return static_cast<int64_t>(pmccntr) * 64;  // Should optimize to << 6
++    }
++  }
++  struct timeval tv;
++  gettimeofday(&tv, nullptr);
++  return static_cast<int64_t>(tv.tv_sec) * 1000000 + tv.tv_usec;
++}
++#else
++#error __ARM_ARCH < 6 does not have a standard cyclecount
++#endif
++#endif
```

Comment:
❤️

Reply:
Keeping the patch aligned with the PR that's over in bwa-mem2 github so trying not to change too much.  But if someone does try to build this on a microcontroller.. good luck!  ;-)